docker/lite/Dockerfile

```diff
@@ -12,29 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# NOTE: We have to build the Vitess binaries from scratch instead of sharing
-#       a base image because Docker Hub dropped the feature we relied upon to
-#       ensure images contain the right binaries.
-
-# Use a temporary layer for the build stage.
-ARG bootstrap_version=34
-ARG image="vitess/bootstrap:${bootstrap_version}-mysql80"
-
-FROM "${image}" AS builder
+FROM --platform=linux/amd64 golang:1.22.5-bullseye AS builder
 
 # Allows docker builds to set the BUILD_NUMBER
 ARG BUILD_NUMBER
 
-# Re-copy sources from working tree.
-COPY --chown=vitess:vitess . /vt/src/vitess.io/vitess
+WORKDIR /vt/src/vitess.io/vitess
 
-# Build and install Vitess in a temporary output directory.
+# Create vitess user
+RUN groupadd -r vitess && useradd -r -g vitess vitess
+RUN mkdir -p /vt/vtdataroot /home/vitess
+RUN chown -R vitess:vitess /vt /home/vitess
 USER vitess
 
+# Re-copy sources from working tree.
+COPY --chown=vitess:vitess . /vt/src/vitess.io/vitess
+
 RUN make install PREFIX=/vt/install
 
 # Start over and build the final image.
-FROM debian:bullseye-slim
+FROM --platform=linux/amd64 debian:bullseye-slim
```

Comment:
I have added these `--platform=linux/amd64` directives here and in other places to ensure that the base image we pull is based on `linux/amd64`, that way, we can build the docker images from any machine. For instance, if we remove this directive and try to build from a Mac, the install dependencies step will fail as there are no Percona package for the Mac's architecture.